packages/@storefront/flux-capacitor/test/unit/core/emitter.ts

```diff
@@ -0,0 +1,293 @@
+import { EventEmitter } from 'eventemitter3';
+import * as sinon from 'sinon';
+import Emitter from '../../../src/core/emitter';
+import Events from '../../../src/core/events';
+import * as utils from '../../../src/core/utils';
+import suite from '../_suite';
+
+suite('Emitter', ({ expect, spy, stub }) => {
+  let emitter;
+
+  beforeEach(() => {
+    emitter = new Emitter();
+  });
+
+  describe('constructor()', () => {
+    it('should extend EventEmitter', () => {
+      expect(emitter).to.be.an.instanceOf(EventEmitter);
+    });
+
+    it('should set _barriers and _lookups', () => {
+      expect(emitter._barriers).to.eql({});
+      expect(emitter._lookups).to.eql({});
+    });
+  });
+
+  describe('all()', () => {
+    let events1;
+    let events2;
+    let key1;
+    let key2;
+    let callback1;
+    let callback2;
+
+    beforeEach(() => {
+      events1 = ['a','b'];
+      events2 = ['a','c'];
+      key1 = 'a\nb';
+      key2 = 'a\nc';
+      callback1 = spy();
+      callback2 = spy();
+    });
+
+    it('should return the emitter' , () => {
+      expect(emitter.all(['a', 'b', 'c'], () => null)).to.eq(emitter);
+    });
+
+    it('should throw an error when `events` is not an array', () => {
+      const error = '`events` is not an array.';
+      const vals = [
+        true,
+        false,
+        0,
+        1,
+        null,
+        undefined,
+        {},
+        () => null,
+        'Hello, world!',
+      ];
+
+      vals.forEach((val) => expect(() => emitter.all(val , () => null)).to.throw(error));
```

Comment:
Pull the `forEach` up and run an `it` for each case. Put the value in the description of the `it`.

Reply:
Implemented.